Modules/_testcapimodule.c

```diff
@@ -5053,6 +5053,96 @@ recurse_infinitely_error_init(PyObject *self, PyObject *args, PyObject *kwds)
 }
 
 
+typedef struct {
+    PyObject_HEAD
+    PyObject *item;
+} PyGenericObject;
+
+
+/* Test PEP 560 slots */
+
+
+static PyObject * generic_new(PyObject *item);
+
+
+static int
+generic_init(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    PyGenericObject *o = (PyGenericObject*) self;
+
+    PyObject *item;
+    if (!PyArg_UnpackTuple(args, "Generic", 1, 1, &item)) {
+        return -1;
+    }
+
+    o->item = item;
```

Comment:
Is there a DECREF missing here? [PyArg_UnpackTuple returns a borrowed item](https://docs.python.org/3/c-api/arg.html#c.PyArg_UnpackTuple). Also there might be a value already here so a DECREF would also be needed (before the assignment). Unless I misunderstand something (totally possible, I feel very rusty in this area :-).

Reply:
I think `Py_INCREF(item)` is needed.

Reply:
But for the purposes of this test we should be able to just remove this function and `tp_init` slot.